.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,150 @@
+        "eamodio.gitlens",
+        // Docker
+        "ms-azuretools.vscode-docker",
+        // TOML
+        "tamasfe.even-better-toml",
+        // YAML
+        "redhat.vscode-yaml",
+        // Spell checking
+        "streetsidesoftware.code-spell-checker",
+        // Resource monitor
+        "mutantdino.resourcemonitor",
+        // Path / filename autocomplete
+        "ionutvmi.path-autocomplete"
+      ]
+    }
+  },
+  // Use 'forwardPorts' to make a list of ports inside the container available locally.
+  "forwardPorts": [5901, 6080],
+  "portsAttributes": {
+    "5901": {
+      "label": "vnc"
+    },
+    "6080": {
+      "label": "vnc web"
+    }
+  },
+  // Use 'postCreateCommand' to run commands after the container is created.
+  "postCreateCommand": {
+    // Initialize shells to use conda and mamba
+    "init-mamba-shell": "mamba init --quiet zsh",
+    // Force pip to always install "globally": This will work in a virtual environment, but fail with the system Python, which is
+    // precicely what we want: prevent users from accidentally cluttering their user site-packages folder
+    "disable-pip-user-installs": "pip config set install.system true",
+    // Install MNE in editabe mode, and set up pre-commit hooks
+    "install-mne-editable": "pip install -e '.[full,dev,test_extra]' && pre-commit install",
+    // Create XDG_RUNTIME_DIR
+    "create-xdg-runtime-dir": "mkdir -p -m 0700 $XDG_RUNTIME_DIR"
+  },
+  // Set `remoteUser` to `root` to connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
+  "remoteUser": "vscode",
+  "postStartCommand": {
+    // Do all git-config-related things here, otherwise we may run into problems; see:
+    // https://github.com/microsoft/vscode-remote-release/issues/6810
+    //
+    // Set git default branch name
+    "set-git-default-branch-name": "git config --global init.defaultBranch main",
+    // Work around "dubious ownership in repository" error
+    "fix-git-ownership-issue": "git config --global --add safe.directory ${containerWorkspaceFolder}",
+    // Use VS Code as default git editor, diff, and merge tool
+    "set-vscode-as-git-editor": "git config --global core.editor 'code --wait --reuse-window'",
+    "add-vscode-as-git-difftool": "git config --global --replace-all difftool.default-difftool.cmd 'code --wait --diff $LOCAL $REMOTE'",
+    "set-vscode-as-git-difftool": "git config --global --replace-all diff.tool default-difftool",
+    "add-vscode-as-git-mergetool": "git config --global --replace-all mergetool.code.cmd 'code --wait --merge $REMOTE $LOCAL $BASE $MERGED'",
+    "set-vscode-as-git-mergetool": "git config --global --replace-all merge.tool code",
+    // Show indicator for "dirty" git repositories in the shell prompt.
+    // This can be slow on large repositories and should be disabled in that case.
+    "enable-dirty-git-repo-indicator": "git config --global --replace-all devcontainers-theme.show-dirty 1"
```

Comment:
is MNE not a "large repository"? I would have thought it was (but I've not personally tried this setting before)

Reply:
Haha no, no, MNE is tiny :) I've never run into issues so far.

I think I did have problems with this setting working on a network filesystem, creating thousands of big files (converted about of TB of MEG data to BIDS). Then everything started stalling.

But during MNE dev, I never had this issue.

Reply:
I added the comment above this setting to make it easy to find out what to change in case we ever run into problems. Marking as resolved for now